tokio-timer/src/error.rs

```diff
@@ -0,0 +1,23 @@
+use self::Kind::*;
+
+/// Errors that occur when working with a timer.
+#[derive(Debug)]
+pub struct Error(Kind);
+
+#[derive(Debug)]
+enum Kind {
+    Shutdown,
+    AtCapacity,
+}
+
+impl Error {
+    /// Create an error representing a shutdown timer.
+    pub fn shutdown() -> Error {
+        Error(Shutdown)
+    }
+
+    /// Create an error representing a timer at capacity.
+    pub fn at_capacity() -> Error {
+        Error(AtCapacity)
+    }
```

Comment:
Should there be inspection methods also? Should a user want to act different depending on if the timer was shutdown (permanent error) vs at capacity (temporary)?

Reply:
Ah yes, you are correct. This is an omission.

Reply:
This is done.